### ISSUE TITLE
gcp-cups-connector Segmentation fault on jessie 64-bit

### ISSUE THREAD

**Issue by jamesshoemaker** [google/cups-connector#159]
I downloaded the binary from https://github.com/google/cups-connector/releases/download/2015.10.05/gcp-cups-connector-linux-amd64-2015.10.05.tar.gz

extracted it, ran 
./gcp-cups-connector-util -init

answered the questions, then tried to run the connector:

./gcp-cups-connector -config-filename gcp-cups-connector.config.json  -logtostderr 
E1227 13:27:34.648403   25658 gcp-cups-connector.go:34] Google Cloud Print CUPS Connector version 2015.10.05-linux
Google Cloud Print CUPS Connector version 2015.10.05-linux
Segmentation fault

without the config file it segfaults, just faster.

**Comment by jay0lee** [comment]
Can you confirm you followed the steps at:

https://github.com/google/cups-connector/wiki/Install

**Comment by jamesshoemaker** [comment]
I'm running debian Jessie and I

sudo apt-get install libcups2 libavahi-client3 avahi-daemon git bzr

then I downloaded the gcp-cups-connector-linux-amd64-2015.10.05.tar.gz
file, extracted it and attempted to run it:

 ./gcp-cups-connector
E1229 13:15:15.237186   10049 gcp-cups-connector.go:34] Google Cloud
Print CUPS Connector version 2015.10.05-linux
Google Cloud Print CUPS Connector version 2015.10.05-linux
Segmentation fault

cat /etc/cups/client.conf
Encryption Never
ServerName fs.dhlake.com

  any other hints?

**Comment by komasoftware** [comment]
Did you follow this guide : https://github.com/google/cups-connector/wiki/Installing-on-Raspberry-Pi-Raspbian-Jessie ?
Works like a charm for me on Raspbian Jessie light. 
```
pi@raspberrypi:~ $ sudo systemctl status gcp
● gcp.service - Google Cloud Print CUPS Connector
   Loaded: loaded (/etc/systemd/system/gcp.service; enabled)
   Active: active (running) since Wed 2015-12-30 00:23:37 CET; 13h ago
 Main PID: 888 (gcp-cups-connec)
   CGroup: /system.slice/gcp.service
           └─888 /opt/gcp_cups_connector/gcp-cups-connector -config-filename /etc/gcp_cups_connector/gcp-cups-connector.config.json

Dec 30 00:23:37 raspberrypi systemd[1]: Started Google Cloud Print CUPS Connector.
Dec 30 00:23:37 raspberrypi gcp-cups-connector[888]: E1230 00:23:37.628646     888 gcp-cups-connector.go:34] Google Cloud Print CUPS Connector version 2015.10.05-linux
Dec 30 00:23:37 raspberrypi gcp-cups-connector[888]: Google Cloud Print CUPS Connector version 2015.10.05-linux
Dec 30 00:23:42 raspberrypi gcp-cups-connector[888]: E1230 00:23:42.663721     888 gcp-cups-connector.go:141] Ready to rock as proxy 'KOMA-PRINT' and in local mode
Dec 30 00:23:42 raspberrypi gcp-cups-connector[888]: Ready to rock as proxy 'KOMA-PRINT' and in local mode
```

This is a 32-bit system though : 

``
pi@raspberrypi:~ $ file /lib/systemd/systemd
/lib/systemd/systemd: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.32, BuildID[sha1]=f11f9dbb6fe628c5dfad514d8de47cab8ffb3c09, stripped
``

**Comment by jamesshoemaker** [comment]
I followed those steps except for the part creating a service, the
program won't even run from the command line.  Not sure why.  I'd build
from source, but jessie's go is older than what the source says is required.

**Comment by jacobmarble** [comment]
Thinking. What is the output of this command?
```
ldd gcp-cups-connector
```

**Comment by jamesshoemaker** [comment]
ldd gcp-cups-connector
        linux-vdso.so.1 (0x00007ffc8b9aa000)
        libcups.so.2 => /usr/lib/x86_64-linux-gnu/libcups.so.2
(0x00007f9be54ac000)
        libavahi-client.so.3 =>
/usr/lib/x86_64-linux-gnu/libavahi-client.so.3 (0x00007f9be529b000)
        libavahi-common.so.3 =>
/usr/lib/x86_64-linux-gnu/libavahi-common.so.3 (0x00007f9be508e000)
        libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
(0x00007f9be4c93000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0
(0x00007f9be4a76000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9be46cd000)
        libgssapi_krb5.so.2 =>
/usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f9be4482000)
        libgnutls-deb0.so.28 =>
/usr/lib/x86_64-linux-gnu/libgnutls-deb0.so.28 (0x00007f9be4163000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f9be3f48000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9be3c47000)
        libdbus-1.so.3 => /lib/x86_64-linux-gnu/libdbus-1.so.3
(0x00007f9be39ff000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9be37fb000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9be5724000)
        libkrb5.so.3 => /usr/lib/x86_64-linux-gnu/libkrb5.so.3
(0x00007f9be3527000)
        libk5crypto.so.3 => /usr/lib/x86_64-linux-gnu/libk5crypto.so.3
(0x00007f9be32f6000)
        libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2
(0x00007f9be30f2000)
        libkrb5support.so.0 =>
/usr/lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f9be2ee6000)
        libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1
(0x00007f9be2ce2000)
        libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2
(0x00007f9be2acb000)
        libp11-kit.so.0 => /usr/lib/x86_64-linux-gnu/libp11-kit.so.0
(0x00007f9be2885000)
        libtasn1.so.6 => /usr/lib/x86_64-linux-gnu/libtasn1.so.6
(0x00007f9be2671000)
        libnettle.so.4 => /usr/lib/x86_64-linux-gnu/libnettle.so.4
(0x00007f9be243f000)
        libhogweed.so.2 => /usr/lib/x86_64-linux-gnu/libhogweed.so.2
(0x00007f9be2210000)
        libgmp.so.10 => /usr/lib/x86_64-linux-gnu/libgmp.so.10
(0x00007f9be1f8d000)
        libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6
(0x00007f9be1d85000)

**Comment by jacobmarble** [comment]
I seriously dropped the ball here. The segfault also occurs on my own jessie workstation. Another user is also confirming this behavior via email.

I'm working on a new release, should be out by Tuesday. I'll be certain to:
- Test on jessie, create special build if needed.
- Add a note about exactly which distro/version I have tested.

**Comment by jamesshoemaker** [comment]
Glad it's reproducible on your local install, as a developer I know how
frustrating it is to have users report things I can't recreate.  Looking
forward to the new release.

James

**Comment by jacobmarble** [comment]
I released 2016.01.02 today. Re-open this issue if that doesn't work.
